### DISCUSSION TITLE
My pages are not getting cached (Next 13.2/13.3 App Router)

### DISCUSSION THREAD

**Discussion by sandervspl** [vercel/next.js#48494] [Help] [answered]
### Summary

For some reason, my pages are not getting cached properly anymore. I pushed an update to my website to test out the new `revalidate` prop on fetch. This increased my bandwidth **by up to 80x** so I removed all `revalidate` and `cache` props from my fetch requests. Even after all that I still can not seem to get my website to cache pages properly anymore. I've tried downgrading to 13.1 but that didn't fix it either. I have done a full code search on `revalidate` and `cache` and I can confirm there are none left.

I have added `export const revalidate = 86400;` but that didn't change anything.

I've also tried `export const dynamic = 'force-static'` but that completely killed my ISR from working. My site has has a lot of content, so I only pre-build the most popular pages (which are the `/mm` pages). All other pages, such as `/2v2` need to be build on request. This stops working when I add `dynamic = 'force-static'` to `page.tsx`. I can also see in the terminal that all prefetches return a 404 for every page it tries to prefetch. You can see that happening on this environment https://wowvalor-git-develop-fix-sandervspl-team.vercel.app/en

Example page that should be build on request: https://wowvalor-git-develop-fix-sandervspl-team.vercel.app/en/death-knight/blood/2v2

I've also tried using `React.cache` for my fetch/db requests but that causes another bug to appear, which is that every page now returns RSC data instead of my page. You can see that bug appear on this environment: https://wowvalor-git-develop-sandervspl-team.vercel.app/en And on this page in particular: https://wowvalor-git-develop-sandervspl-team.vercel.app/en/death-knight/blood/mm

So in short: I need my pages to be cached again and I don't know how.
- I don't use dynamic pages
- I don't use the new `revalidate` fetch prop
- I don't change the `cache` fetch prop
- I cannot use `export const dynamic = 'force-static'` because it breaks my site
- I cannot use `React.cache` because it also breaks my site

These graphs show the huge difference in caching.

**Before**
<img width="1198" alt="Screenshot 2023-04-17 at 18 43 35" src="https://user-images.githubusercontent.com/4492636/232553924-dddcaf63-1a23-4ae1-a343-2c94b2aa79d4.png">

[this](https://wowvalor-o4aoksx98-sandervspl-team.vercel.app/en/monk/windwalker/mm) is a version I found that is still working and if you open up the terminal, go to `network` and filter on `mm` you can open the dropdown in the `monk` drop down and hover over any of the items. You should see the site prefetch the other pages. These should all return with `x-vercel-cache: HIT`.

**After**
<img width="1196" alt="Screenshot 2023-04-17 at 18 43 14" src="https://user-images.githubusercontent.com/4492636/232553918-158ddc3a-9879-44c4-aad0-c428e8ac1fec.png">

Do the same on [the production site](https://www.wowvalor.app/en/monk/windwalker/mm) and you'll see it will return `x-vercel-cache: STALE`.

### Additional information

- This project uses the App Router
- I have tried these versions: 13.1.x, 13.2.x, 13.3.0, and 13.3.1-canary.x
- It uses the Supabase JS SDK to fetch from the Supabase DB. The SDK does not use a custom fetch function
- It uses Node 18 and pnpm 8
- I don't use `headers()`, `cookies()` or `useSearchParams()`
- I don't use `React.suspense`

**Comment by sandervspl** [comment, marked as answer]
I have finally found the solution. I was using Edge Config in some components. For some reason this broke the caching. I have used it since its inception, so I'm not sure what went wrong. I removed it now and I can see that more than 85% of my requests are cached again.

**Comment by hjaber** [comment]
Your answer helped me troubleshoot a similar issue, thank you for your detailed post.

I have opened issue #48889
